app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsViewModel.kt

```diff
@@ -33,68 +36,30 @@ class StatisticsViewModel(
     private val tasksRepository: TasksRepository
 ) : ViewModel() {
 
-    private val _dataLoading = MutableLiveData<Boolean>()
-    val dataLoading: LiveData<Boolean> = _dataLoading
-
-    private val _error = MutableLiveData<Boolean>()
-    val error: LiveData<Boolean> = _error
-
-    /**
-     * Controls whether the stats are shown or a "No data" message.
-     */
-    private val _empty = MutableLiveData<Boolean>()
-    val empty: LiveData<Boolean> = _empty
-
-    private val _activeTasksPercent = MutableLiveData<Float>()
-    val activeTasksPercent: LiveData<Float> = _activeTasksPercent
-
-    private val _completedTasksPercent = MutableLiveData<Float>()
-    val completedTasksPercent: LiveData<Float> = _completedTasksPercent
-
-    private var activeTasks = 0
-
-    private var completedTasks = 0
-
-    init {
-        start()
+    private val tasks: LiveData<Result<List<Task>>> = tasksRepository.observeTasks()
+    private val _dataLoading = MutableLiveData<Boolean>(false)
+    private val stats: LiveData<StatsResult?> = tasks.map {
+        if (it is Success) {
+            getActiveAndCompletedStats(it.data)
+        } else {
+            null
+        }
     }
 
-    fun start() {
-        if (_dataLoading.value == true) {
-            return
-        }
-        _dataLoading.value = true
+    val activeTasksPercent = stats.map {
+        it?.activeTasksPercent ?: 0f }
+    val completedTasksPercent: LiveData<Float> = stats.map { it?.completedTasksPercent ?: 0f }
+    val dataLoading: LiveData<Boolean> = _dataLoading
+    val error: LiveData<Boolean> = tasks.map { it is Error }
+    val empty: LiveData<Boolean> = tasks.map { (it as? Success)?.data.isNullOrEmpty() }
 
+    fun refresh() {
+        _dataLoading.value = true
         wrapEspressoIdlingResource {
```

Comment:
Why is this used here and nowhere else? There's very similar code in the other viewmodels for loading data.

Reply:
Looks like _dataLoading is not used in the VM, I guess because of some merge issue. Something for another PR.